### PR TITLE
GEODE-4370: Ensure that rmi-io uses the JMX/RMI port

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/GeodeRemoteStreamExporter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.rmi.RemoteException;
+import java.rmi.server.RMIClientSocketFactory;
+import java.rmi.server.RMIServerSocketFactory;
+import java.rmi.server.UnicastRemoteObject;
+
+import com.healthmarketscience.rmiio.RemoteStreamServer;
+import com.healthmarketscience.rmiio.exporter.DefaultRemoteStreamExporter;
+
+public class GeodeRemoteStreamExporter extends DefaultRemoteStreamExporter {
```
Comment (metatype) [inline geode-core/src/main/java/org/apache/geode/management/internal/GeodeRemoteStreamExporter.java]: Would it be cleaner to extend `RemoteStreamExporter` directly?  That avoids the conflict between the `_port` and `port` member variables.

Comment (jdeppe-pivotal) [reply geode-core/src/main/java/org/apache/geode/management/internal/GeodeRemoteStreamExporter.java]: Yes, that would work too.

File: geode-core/src/main/java/org/apache/geode/management/internal/ManagementAgent.java [hunk 2]
```diff
@@ -546,6 +550,10 @@ public JMXConnectorServer getJmxConnectorServer() {
     return jmxConnectorServer;
   }
 
+  public RemoteStreamExporter getRemoteStreamExporter() {
+    return new GeodeRemoteStreamExporter(port, rmiClientSocketFactory, rmiServerSocketFactory);
```
Comment (jinmeiliao) [inline geode-core/src/main/java/org/apache/geode/management/internal/ManagementAgent.java]: do we need to create a new exporter every time we get it? Seems like RemoteStreamExporter.getInstance() will return the same instance every time.

**Comment by jdeppe-pivotal** [comment]
precheckin is green

**Comment by metatype** [comment]
One more question:  if _User A_ does `gfsh deploy` and another _User B_ is issuing gfsh commands at the same time what will happen?  Will the `exportObject` cause concurrent gfsh / jmx requests to fail?

**Comment by jinmeiliao** [comment]
what would that be different than 2 users executing any gfsh commands at the same time using different gfsh clients?

**Comment by jdeppe-pivotal** [comment]
@metatype `exportObject` exports individual instances of objects (and returns those for clients to use). Thus each client will use a uniquely exported object. As an aside, the object is only exported for the lifetime of the method invocation (in this case). Once a deploy is complete, the necessary objects are unexported.

**Pull request merged**